Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
-    int fillX = m_LastDoorMaterialPos.GetFloorIntX();
-    int fillY = m_LastDoorMaterialPos.GetFloorIntY();
-
-    if (g_SceneMan.GetTerrMatter(fillX, fillY) != g_MaterialAir)
-    {
-        floodfill(g_SceneMan.GetTerrain()->GetMaterialBitmap(), fillX, fillY, g_MaterialAir);
-        // Register that we changed the material layer of the terrain
-        g_SceneMan.GetTerrain()->AddUpdatedMaterialArea(m_pDoor->GetBoundingBox());
-        return true;
-    }
-
-    return false;
-}
-
-} // namespace RTE
\ No newline at end of file
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool ADoor::EraseDoorMaterial() {
+		m_DoorMaterialDrawn = false;
+
+		if (!m_Door || !g_SceneMan.GetTerrain() || !g_SceneMan.GetTerrain()->GetMaterialBitmap()) {
+			return false;
+		}
+
+		int fillX = m_LastDoorMaterialPos.GetFloorIntX();
+		int fillY = m_LastDoorMaterialPos.GetFloorIntY();
+
+		if (g_SceneMan.GetTerrMatter(fillX, fillY) != g_MaterialAir) {
+			floodfill(g_SceneMan.GetTerrain()->GetMaterialBitmap(), fillX, fillY, g_MaterialAir);
+			// Register that we changed the material layer of the terrain
+			g_SceneMan.GetTerrain()->AddUpdatedMaterialArea(m_Door->GetBoundingBox());
+			return true;
+		}
+		return false;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::MaterialDrawOverride(bool enable) {
+		if (!g_SceneMan.GetTerrain() || !g_SceneMan.GetTerrain()->GetMaterialBitmap()) {
+			return;
+		}
+		if (enable) {
+			// Erase the material temporarily if we have drawn it and the override isn't already in effect
+			if (m_DoorMaterialDrawn && m_MaterialDrawOverride != enable) {
+				int fillX = m_LastDoorMaterialPos.GetFloorIntX();
+				int fillY = m_LastDoorMaterialPos.GetFloorIntY();
+
+				if (g_SceneMan.GetTerrMatter(fillX, fillY) != g_MaterialAir) {
+					floodfill(g_SceneMan.GetTerrain()->GetMaterialBitmap(), fillX, fillY, g_MaterialAir);
+					// Register that we changed the material layer of the terrain
+					g_SceneMan.GetTerrain()->AddUpdatedMaterialArea(m_Door->GetBoundingBox());
+				}
+			}
+			m_MaterialDrawOverride = enable;
+		} else {
+			// Draw the door back if we were indeed temporarily suppressing it before
+			if (m_DoorMaterialDrawn && m_MaterialDrawOverride != enable) {
+				Vector notUsed;
+				m_Door->Draw(g_SceneMan.GetTerrain()->GetMaterialBitmap(), notUsed, g_DrawMaterial, true);
+				// Register that we changed the material layer of the terrain
+				g_SceneMan.GetTerrain()->AddUpdatedMaterialArea(m_Door->GetBoundingBox());
+			}
+			m_MaterialDrawOverride = enable;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int ADoor::RemoveAnyRandomWounds(int amount) {
+		float damage = 0;
+		for (int i = 0; i < amount; i++) {
+			// Fill the list of damaged bodyparts
```

Comment:
Rename bodyParts to woundedBodyParts so you can kill all these comments.
Also
1. If bodyParts.szie() == 0 there's no reason to keep trying to remove wounds cause there are clearly none, just return damage then and there
2. If you felt like it, this whole thing could be made more efficient by not remaking the vector of wounded body parts each time:
* Make it once and keep a count of how many wounds are left on each wounded body part. Something like `std::vector<std::pair<MOSRotating *, std::size_t>> woundedBodyParts; woundedBodyParts.push_back({this, GetWoundCount()});`
* When you remove a wound, lower your wound count for the wounded body part by 1
* If a wounded body part no longer has wounds left (i.e. your count is 0), remove its entry from the list of wounded body parts.
* Everything else is the same

Reply:
Feeling extra big dum so i'll let you deal with this one.

Reply:
Okay, as mentioned in discord, I'll take a look after all your other stuff is done. Let me know when that is.

Reply:
This will be dealt with in #143.